### PR TITLE
Update fetch tests for more lenient Request constructor

### PR DESCRIPTION
**Pull request by jdm**
This addresses the tests affected by https://github.com/whatwg/fetch/pull/377.

### CODE REVIEW

**Comment by jdm** [comment]
Both Firefox and Chrome fail the new tests, presumably because their implementation haven't updated to match https://github.com/whatwg/fetch/pull/377.

**Comment by jdm** [comment]
Firefox's bug is https://bugzilla.mozilla.org/show_bug.cgi?id=1298823 .

File: fetch/api/request/request-error.html [hunk 1]
```diff
@@ -37,8 +37,8 @@
       },"RequestInit's referrer is invalid");
 
       test(function() {
-        assert_throws(new TypeError() , function() { new Request("", {"referrer" : "http://test.url"}); },
-          "Expect TypeError exception");
+        var req = new Request("", {"referrer" : "http://test.url"});
+        assert_equals(req.referrer, "about:client", "cross-origin referrer should become client referrer");
```
Comment (youennf) [inline fetch/api/request/request-error.html]: Ditto here, search for "var referrers ="

File: fetch/api/request/request-error.html [hunk 2]
```diff
@@ -27,8 +27,8 @@
       },"Input URL has credentials");
 
       test(function() {
-        assert_throws(new TypeError() , function() { new Request("", {"mode" : "navigate"}); },
-          "Expect TypeError exception");
+        var req = new Request("", {"mode" : "navigate"});
+        assert_equals(req.mode, "same-origin", "navigate mode in initializer should become same-origin");
```
Comment (youennf) [inline fetch/api/request/request-error.html]: I would remove that test and update fetch/api/request/request-init-001.sub.html instead (check for "var modes =")

**Comment by jdm** [comment]
Review comments addressed.

**Review by youennf — APPROVED** [review]

**Comment by jdm** [comment]
@jgraham Do you know what to make of that travis failure?

**Pull request merged**